module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeTagTest.php

```diff
@@ -0,0 +1,193 @@
+            '<button class="btn" id="login">text</button>',
+            $helper('button', 'text', ['class' => 'btn', 'id' => 'login'])
+        );
+
+        // String
+        $this->assertEquals(
+            '<i class="btn">text</i>',
+            $helper('i', 'text', 'btn')
+        );
+
+        // Empty text
+        $this->assertEquals(
+            '<i class="fa&#x20;fa-awesome"></i>',
+            $helper('i', '', 'fa fa-awesome')
+        );
+
+        // Truthy attribute
+        $this->assertEquals(
+            '<a href="&#x2F;login" data-lightbox="1">Login</a>',
+            $helper('a', 'Login', ['href' => '/login', 'data-lightbox' => true])
+        );
+    }
+
+    /**
+     * Test escapeContent
+     */
+    public function testOptionEscape()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+            )
+        );
+
+        $this->assertEquals(
+            '<button>This link is <strong>important</strong></button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+                [],
+                ['escapeContent' => false]
+            )
+        );
+
+        $this->assertEquals(
+            '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+                [],
+                ['escapeContent' => true]
+            )
+        );
+    }
+
+    /**
+     * Test escapeContent
+     */
+    public function testVoidElements()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<img src="book.gif" />',
+            $helper(
+                'img',
+                '',
+                ['src' => 'book.gif']
+            )
+        );
+
+        // Class only
+        $this->assertEquals(
+            '<br class="sm&#x3A;hidden" />',
+            $helper(
+                'br',
+                '',
+                'sm:hidden'
+            )
+        );
+    }
+
+    /**
+     * Test tag name edge cases
+     */
+    public function testValidTagNames()
+    {
+        $helper = $this->getHelper();
+
+        $helper('CAPITAL', '');
+        $helper('mIxEdCaSe', '');
+        $helper('my-custom', '');
+        $helper('my-long-custom', '');
+        $helper('is---this---ok', '');
+    }
+
+    /*
+     * Bad tag names for test below
+     */
+    public function invalidTags()
+    {
+        return [
+            ['n0numbers'],
+            ['-must-start-with-letter'],
+            ['em—dash'],
+            ['<doubleangles>'],
+            ['?php'],
+        ];
+    }
+
+    /**
+     * Test exception on bad tag names
+     *
+     * @dataProvider invalidTags
```

Comment:
I wonder if some of the earlier tests would also benefit from `@dataProvider` usage.

Reply:
I prefer the current style, but I'd be happy to migrate to this format.

Reply:
The main advantage to the dataProvider approach is that it gives us more specific information when things fail. Some of these tests that include many assertions could be slightly annoying to troubleshoot if they fail in the future, but if we use a data provider, it will tell us exactly which case is failing (especially if we use the associative array format, where key is description and value is data to provide).

I'll leave it to your discretion, though; if you don't want to spend more time revising this, we can always refactor later!